### PR TITLE
Create and use minimal interface target for AdePT

### PR DESCRIPTION
**Pull request by drbenmorgan**
Add a CMake `INTERFACE` target for AdePT with main locations of headers and dependency on CopCore as usage requirements. This allows all of the examples and tests to use `target_link_libraries` to pick up all the headers (and pave the way to simplify the repo layout in a separate PR), and these have been updated accordingly.

The AdePT target is not totally complete in that it doesn't declare its dependence on VecGeom (through `LoopNavigator.h` in the base, and others in magneticfield/tracking/physics) yet. That's deliberate as how to correctly declare the CUDA vs CPU only usage requirements needs some thought, but the target as implemented here is a good first step.

### CODE REVIEW

File: examples/Example5/CMakeLists.txt [hunk 1]
```diff
@@ -1,14 +1,10 @@
 # SPDX-FileCopyrightText: 2020 CERN
 # SPDX-License-Identifier: Apache-2.0
 
-if(NOT TARGET g4HepEm)
+if(NOT TARGET G4HepEm::g4HepEm)
```
Comment (hahnjo) [inline examples/Example5/CMakeLists.txt:4]: Is this needed? I think we should use the target consistently, either with or without scope (see below and `example6`).

Comment (drbenmorgan) [reply examples/Example5/CMakeLists.txt:4]: Good point on the scope, I've changed the remainder of G4HepEm consumers to use the scoped target names.

**Review by hahnjo — APPROVED** [review]
I like this :+1: (not being a CMake expert)

**Review by agheata — APPROVED** [review]
Nice!

**Pull request merged**